@here/harp-mapview/lib/GeoJsonTiler.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2018-2019 HERE Global B.V. and its affiliate(s). All rights reserved.
+ *
+ * This software and other materials contain proprietary information controlled by HERE and are
+ * protected by applicable copyright legislation. Any use and utilization of this software and other
+ * materials and disclosure to any third parties is conditional upon having a separate agreement
+ * with HERE for the access, use, utilization or disclosure of this software. In the absence of such
+ * agreement, the use of the software is not allowed.
+*/
+
+import { GeoJson, ITiler } from "@here/harp-datasource-protocol";
+import { TileKey } from "@here/harp-geoutils";
+import {LoggerManager} from "../../harp-utils/lib/Logger";
+
+// tslint:disable-next-line:no-var-requires
+const geojsonvt = require("geojson-vt");
+
+const logger = LoggerManager.instance.create("GeoJsonTiler");
+
+export class GeoJsonTiler implements ITiler {
+    indexes: Map<string, ITiler>;
+
+    constructor() {
+        this.indexes = new Map();
+    }
+
+    dispose() {/* */
+    }
+
+    async connect(): Promise<void> {
+        return Promise.resolve();
+    }
+
+    async registerIndex(indexId: string, input: URL | GeoJson): Promise<void> {
+        if (this.indexes.has(indexId)) { return; }
+        if (input instanceof URL) {
+            const response = await fetch(input.href);
+            if (!response.ok) {
+                logger.error(`${input} Status Text:  ${response.statusText}`);
+                return Promise.reject();
```

Comment:
Don't `log` error, but throw it so it's caller responsibility to handle and/or log.
Like this:
```
throw new Error(`GeoJsonTiler: Unable to fetch ${url}: ${response.statusText}`)
```